mod-users/src/main/resources/domain/mod-users/features/users.feature

```diff
@@ -0,0 +1,111 @@
+    And match response.users[0].personal.firstName == expectedFirstName
+
+  Scenario: Search user by firstname & lastname.
+    * def expectedFirstName = 'first'
+    * def expectedLastName = 'TestUser'
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def responseBarcode = createUserResponse.response.firstName
+    Given path 'users?query=(personal.firstName=first)or(personal.lastName=TestUser)'
+    When method GET
+    Then status 200
+    And match response.users[0].personal.firstName == expectedFirstName
+    And match response.users[0].personal.lastName == expectedLastName
+
+  Scenario: Search user by UUID.
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def responseUserId = createUserResponse.response.id
+
+    Given path 'users?query=(id='+responseUserId+')'
+    When method GET
+    Then status 200
+    And match responseUserId == response.users[0].id
+
+  Scenario: Search user by lastname.
+    * def expectedLastName = 'TestUser'
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def responseUserLastName = createUserResponse.response.lastName
+
+    Given path 'users?query=(personal.lastName=TestUser)'
+    When method GET
+    Then status 200
+    And match responseUserLastName == response.users[0].lastName
+
+  Scenario: Search user by email.
+    * def expectedEmail = 'testmail@abc.com'
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def responseUserEmail = createUserResponse.response.email
+
+    Given path 'users?query=(personal.email=testmail@abc.com)'
+    When method GET
+    Then status 200
+    And match responseUserEmail == response.users[0].email
+
+  Scenario: Search user by username.
+    * def expectedEmail = 'testmail@abc.com'
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def responseUserEmail = createUserResponse.response.email
+
+    Given path 'users?query=(username=test-user)'
+    When method GET
+    Then status 200
+
+  Scenario: Filter inactive patron.
+    * call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { status: true }
+    * call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { status: true }
+    * call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { status: false }
+    * call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { status: false }
+
+    Given path 'users?query=(active==false)'
+    When method GET
+    Then status 200
+    And match response.resultInfo.totalRecords == 2
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
Why is there a bunch of white space at the bottom of this file?

Reply:
i will remove them , those are automatically saved .